yapapi/engine.py

```diff
@@ -564,9 +653,14 @@ async def worker_task(agreement: Agreement):
             work_context = WorkContext(activity, agreement, self.storage_manager, emitter=job.emit)
             work_context.emit(events.ActivityCreated)
 
+            self._activity_created_at[activity.id] = datetime.now()
+
             async with activity:
                 self.accept_debit_notes_for_agreement(job.id, agreement.id)
                 await run_worker(work_context)
+                # Providers may issue debit notes after activity ends.
+                # This will prevent terminating agreements when this happens.
```

Comment:
Hmm. Let's say our activity ends and from now on our provider sends us a micro-debitnote for a part of the missing amount each second. Why do we want to accept this?

I understand that current `yagna` will send at most a single debit note after the agreement ends (I think?), but is this a good enough reason?

Reply:
yeah, I don't think this is a good reason and I believe we should just ignore such a debit note but don't break the agreement... changed the implementation to do so